pkg/cstor/cvr_operation.go

```diff
@@ -67,12 +75,61 @@ func (p *Plugin) waitForAllCVR(vol *Volume) error {
 		}
 		time.Sleep(CVRCheckInterval)
 	}
+	return errors.Errorf("CVR for volume{%s} are not ready!", vol.volname)
+}
+
+// waitForCSIBasedCVRs will ensure that all CVRs related to
+// the given CSI volume is created.
+func (p *Plugin) waitForCSIBasedCVRs(vol *Volume, replicaCount int) error {
+	for cnt := 0; cnt < CVRWaitCount; cnt++ {
+		cvrList, err := p.OpenEBSAPIsClient.
+			CstorV1().
+			CStorVolumeReplicas(p.namespace).
+			List(metav1.ListOptions{
+				LabelSelector: "openebs.io/persistent-volume=" + vol.volname,
+			})
+		if err != nil {
+			return errors.Errorf("Failed to fetch CVR for volume=%s %s", vol.volname, err)
+		}
+
+		if len(cvrList.Items) != replicaCount {
+			time.Sleep(CVRCheckInterval)
```

Comment:
not related to this PR, can we add a print for every 5 mins? but, not a blocking comment for the merge

Reply:
ok.. this is also not required as this loop won't go for more than 500seconds